src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs

```diff
@@ -79,18 +79,23 @@ public JwtPayload(string issuer, string audience, IEnumerable<Claim> claims, Dat
         /// <param name="issuer">If this value is not null, a { iss, 'issuer' } claim will be added, overwriting any 'iss' claim in 'claims' if present.</param>
         /// <param name="audience">If this value is not null, a { aud, 'audience' } claim will be added, appending to any 'aud' claims in 'claims' if present.</param>
         /// <param name="claims">If this value is not null then for each <see cref="Claim"/> a { 'Claim.Type', 'Claim.Value' } is added. If duplicate claims are found then a { 'Claim.Type', List&lt;object&gt; } will be created to contain the duplicate values.</param>
-        /// <param name="notBefore">If notbefore.HasValue a { nbf, 'value' } claim is added, overwriting any 'nbf' claim in 'claims' if present.</param>
-        /// <param name="expires">If expires.HasValue a { exp, 'value' } claim is added, overwriting any 'exp' claim in 'claims' if present.</param>
-        /// <param name="issuedAt">If issuedAt.HasValue is 'true' a { iat, 'value' } claim is added, overwriting any 'iat' claim in 'claims' if present.</param>
+        /// <param name="claimsCollection">If both claims and claimsCollection are not null then the values in claims will be combined with the values in claimsCollection. The values found in claimCollection take precedence over those found in claims, so any duplicate
+        /// values will be overridden.</param>
+        /// <param name="notBefore">If notbefore.HasValue a { nbf, 'value' } claim is added, overwriting any 'nbf' claim in 'claims' and 'claimcollection' if present.</param>
+        /// <param name="expires">If expires.HasValue a { exp, 'value' } claim is added, overwriting any 'exp' claim in 'claims' and 'claimcollection' if present.</param>
+        /// <param name="issuedAt">If issuedAt.HasValue is 'true' a { iat, 'value' } claim is added, overwriting any 'iat' claim in 'claims' and 'claimcollection' if present.</param>
         /// <remarks>Comparison is set to <see cref="StringComparer.Ordinal"/>
-        /// <para>The 4 parameters: 'issuer', 'audience', 'notBefore', 'expires' take precednece over <see cref="Claim"/>(s) in 'claims'. The values in 'claims' will be overridden.</para></remarks>
+        /// <para>The 4 parameters: 'issuer', 'audience', 'notBefore', 'expires' take precednece over <see cref="Claim"/>(s) in 'claims' and 'claimcollection'. The values will be overridden.</para></remarks>
         /// <exception cref="ArgumentException">If 'expires' &lt;= 'notbefore'.</exception>
-        public JwtPayload(string issuer, string audience, IEnumerable<Claim> claims, DateTime? notBefore, DateTime? expires, DateTime? issuedAt)
+        public JwtPayload(string issuer, string audience, IEnumerable<Claim> claims, IDictionary<string, object> claimsCollection, DateTime? notBefore, DateTime? expires, DateTime? issuedAt)
```

Comment:
Modifying the parameters for this public method would be considered a breaking change. I would make this a new overloaded method.

Reply:
@mafurman @RojaEnnam we are trying to steer users towards JsonWebToken. I am wondering if we should have this method?

All it means is that JwtSecurityTokenHandler consolidates to 'Claim' before calling this method.

Reply:
@brentschmaltz That's actually a good point! I don't think this necessarily needs to be a public API; we can always add it later if it's needed for some reason.

Reply:
@brentschmaltz @mafurman  Originally I implemented it by consolidating dictionary to 'Claim'(s) and then call JwtPayload but, 'AddClaims' method again converts the 'Claim'(s) to dictionary. When this is happening on DateTime objects I observed the original value is not retained.

```
- DateTime.TryParse("2019-11-15T14:31:21.6101326Z") result is "11/15/2019 6:31:21 AM"( which will be the value of Claim after converting dictionary to Claim list)
- After' AddClaims' Date.TryParse(11/15/2019 6:31:21 AM) results in 2019-11-15T14:31:21.0000000Z instead of 019-11-15T14:31:21.6101326Z '

```